qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,324 @@
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import OperatorBase, PauliSumOp, ExpectationBase, CircuitSampler,StateFn
+from qiskit.opflow.gradients import GradientBase, Gradient
+from qiskit.algorithms.minimum_eigen_solvers import MinimumEigensolver
+from qiskit.circuit.library import EvolvedOperatorAnsatz, RealAmplitudes, PauliEvolutionGate
+from qiskit.circuit.library import NLocal
+from qiskit.utils.validation import validate_min
+
+
+from qiskit.algorithms.optimizers import Optimizer
+from qiskit.utils import QuantumInstance
+from qiskit.providers import Backend
+from qiskit.algorithms.aux_ops_evaluator import eval_observables
+#from ..aux_ops_evaluator import eval_observables
+from enum import Enum
+
+# from .minimum_eigensolver_factories import MinimumEigensolverFactory
+# from .ground_state_eigensolver import GroundStateEigensolver
+logger = logging.getLogger(__name__)
+
+
+class Finishing_criterion(Enum):
+    finishing_criterion = ""
+
+
+class AdaptVQE(VariationalAlgorithm):
+    """A ground state calculation employing the AdaptVQE algorithm.
+
+    The performance of AdaptVQE can significantly depend on the choice of gradient method, QFI
+    solver (if applicable) and the epsilon value.
+    """
+
+    def __init__(
+        self,
+        solver: VQE ,
+        ansatz: Optional[QuantumCircuit] = None,
+        threshold: float = 1e-5,
+        max_iterations: Optional[int] = None,
+        adapt_gradient: Optional[GradientBase] = None,
+        initial_point: Optional[np.ndarray] = None,
+        expectation: Optional[ExpectationBase] = None,
+        quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
+        excitation_pool: List[Union[OperatorBase, QuantumCircuit]] = None,
+    ) -> None:
+        """
+        Args:
+            ansatz: A parameterized circuit used as Ansatz for the wave function.
+            solver: a factory for the VQE solver employing a UCCSD ansatz.
+            threshold: the energy convergence threshold. It has a minimum value of 1e-15.
+            max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            gradient: a class that converts operator expression to the first-order gradient based
+                on the method mentioned.
+            initial_point: An optional initial point (i.e. initial parameter values) for the optimizer.
+            excitation_pool: An entire list of excitations.
+        """
+        validate_min("threshold", threshold, 1e-15)
+        super().__init__(
+            initial_point = initial_point
+        )
+
+        if adapt_gradient is None:
+            adapt_gradient = Gradient(grad_method="param_shift")
+        self._threshold = threshold
+        self.solver = solver
+        self._max_iterations = max_iterations
+        self._adapt_gradient = adapt_gradient
+        self._excitation_pool = excitation_pool
+        self._tmp_ansatz = ansatz
```

Comment:
In the past, we would do the following:
```suggestion
        self._tmp_ansatz = copy.deepcopy(ansatz)
```
I am not yet sure if we truly need this